spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala

```diff
@@ -58,6 +58,50 @@ class CometExecSuite extends CometTestBase {
     }
   }
 
+  test("HashJoin without join filter") {
+    withSQLConf(
+      SQLConf.PREFER_SORTMERGEJOIN.key -> "false",
+      SQLConf.ADAPTIVE_AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1",
+      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1") {
+      withParquetTable((0 until 10).map(i => (i, i % 5)), "tbl_a") {
+        withParquetTable((0 until 10).map(i => (i % 10, i + 2)), "tbl_b") {
+          val df1 =
+            sql(
+              "SELECT /*+ SHUFFLE_HASH(tbl_a) */ * FROM tbl_a JOIN tbl_b ON tbl_a._2 = tbl_b._1")
+          checkSparkAnswerAndOperator(df1)
+
+          // TODO: Spark 3.4 returns SortMergeJoin for this query even with SHUFFLE_HASH hint.
+          // We need to investigate why this happens and fix it.
+          /*
+          val df2 =
+            sql("SELECT /*+ SHUFFLE_HASH(tbl_a) */ * FROM tbl_a LEFT JOIN tbl_b ON tbl_a._2 = tbl_b._1")
+          checkSparkAnswerAndOperator(df2)
+
+          val df3 =
+            sql("SELECT /*+ SHUFFLE_HASH(tbl_b) */ * FROM tbl_b LEFT JOIN tbl_a ON tbl_a._2 = tbl_b._1")
+          checkSparkAnswerAndOperator(df3)
```

Comment:
can use this : spark.sql.join.forceApplyShuffledHashJoin for force SHJ always was added in spark to test this joins specifically
https://github.com/apache/spark/pull/33182

Reply:
Let me try. I just wonder why it is not planned as HashJoin by Spark. For right join, it works as expected. Only left join failed.

Reply:
No, it doesn't work. That's something weird. I'd need to look at it further.

Reply:
:( This is strange can for some reason `SPARK_TESTING` might not be set in env ?

Reply:
Oh, this is because left join with build left and right join with build right in hash join is supported in Spark 3.5 or above by https://github.com/apache/spark/pull/41398. Since Comet currently uses 3.4, it isn't supported yet.